src/components/image-tooltip/ImageTooltipContent.tsx

```diff
@@ -3,34 +3,45 @@ import classNames from 'classnames';
 
 import './ImageTooltipContent.scss';
 
+type OnImageLoad = () => void;
+
 export type ImageTooltipContentProps = {
     /** A React component representing the image used in the visual tooltip */
     children: React.ReactElement;
     /** A string to be used in the tooltip's paragraph content */
     content: string;
+    /** A callback triggered onLoad for the image element */
+    onImageLoad?: OnImageLoad;
     /** A string to be used in the tooltip's title heading */
     title: string;
 };
 
-function appendTooltipImageChildClass(child: React.ReactElement): React.ReactElement | null {
+function appendTooltipImageChildClass(child: React.ReactElement, onImageLoad?: OnImageLoad): React.ReactElement | null {
     if (child) {
         const {
             props: { className: existingClasses },
         } = child;
         const className = classNames(existingClasses, 'bdl-ImageTooltipContent-imageChild');
 
-        return React.cloneElement(child, {
-            className,
-        });
+        const propsForElement = onImageLoad
+            ? {
+                  className,
+                  onLoad: onImageLoad,
+              }
+            : {
+                  className,
+              };
+
+        return React.cloneElement(child, propsForElement);
     }
 
     return null;
 }
 
-const ImageTooltipContent = ({ children, content, title }: ImageTooltipContentProps) => (
+const ImageTooltipContent = ({ children, content, onImageLoad, title }: ImageTooltipContentProps) => (
     <div className="bdl-ImageTooltipContent">
         <div className="bdl-ImageTooltipContent-image">
-            {React.Children.map(children, child => appendTooltipImageChildClass(child))}
+            {React.Children.map(children, child => appendTooltipImageChildClass(child, onImageLoad))}
```

Comment:
since there can be multiple children, will this be an issue? should you wait until they are all loaded?

Reply:
looking into this...this might not need to be a map since we only support a single element coming in 🤔